chain-impl-mockchain/src/config.rs

```diff
@@ -224,24 +287,102 @@ impl ConfigParamVariant for ConsensusVersion {
     }
 
     fn to_cfg_string(&self) -> String {
-        format!("{}", self)
+        self.to_string()
     }
 
     fn from_cfg_str(s: &str) -> Result<Self, Error> {
-        s.parse().map_err(|_| Error::UnknownString(s.to_string()))
+        from_cfg_str(s)
     }
 }
 
-#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
-struct Tag(u16);
+impl ConfigParamVariant for LeaderId {
+    fn to_payload(&self) -> Vec<u8> {
+        self.as_ref().to_vec()
+    }
 
-impl Tag {
-    pub const fn new(tag: u16) -> Self {
-        // validate that it's less than 1024 when const fns support ifs
-        Tag(tag)
+    fn from_payload(payload: &[u8]) -> Result<Self, Error> {
+        PublicKey::from_binary(payload)
+            .map(Into::into)
+            .map_err(|_| Error::SizeInvalid)
+    }
+
+    fn to_cfg_string(&self) -> String {
+        self.as_public_key().to_bech32_str()
```

Comment:
I am not sure the encoding in `bech32` holds. Actually this comes to: why this library defines the encoding format as it should be in the `jcli`'s genesis.yaml file, which is application defined.

It seems to me the `to_cfg_string` and `from_cfg_str` functions are not needed at all here.

Reply:
I was thinking about removing `to_cfg_string` and `from_cfg_str` completely. It's only used by `ConfigParam`'s Serde implementation, which we want to get rid of anyway. I guess it's a good moment to purge it.